

package ProtoBuf;

//服务器到客户端的协议号
enum ServerToClientCmd
{
	//登录
	SERVER_TO_CLIENT_LOGIN_BEGIN		= 1;
	LOGIN_TO_CLIENT_CHECK_VERSION		= 4;
	LOGIN_TO_CLIENT_ACCOUNT_LOGIN		= 5;
	LOGIC_TO_CLIENT_PING				= 66;		// 逻辑服ping客户端
    LOGIC_TO_CLIENT_RECONNECT           = 67;       // 断线重连回包
    LOGIC_TO_CLIENT_LOGOUT_LOGICSERVER  = 68;       // 客户端登出逻辑服务器，有可能被挤掉等
	LOGIC_TO_CLIENT_LOGIN_LOGICSERVER	= 69;		// 客户端用账号登录逻辑服务器响应
	LOGIC_TO_CLIENT_GET_CHARACTER_LIST	= 70;		// 请求角色列表响应
	LOGIC_TO_CLIENT_CREATE_CHARACTER	= 71;		// 请求创建角色响应
	LOGIC_TO_CLIENT_DEL_CHARACTER		= 72;		// 删除角色
	LOGIC_TO_CLIENT_NOTIFY_CLIENT_STATE = 73;      // 服务器通知客户端的状态
    SERVER_TO_CLIENT_LOGIN_END			= 90;

	//角色
	SERVER_TO_CLIENT_PLAYER_BEGIN			= 100;
	LOGIC_TO_CLIENT_PLAYERINFO              = 101;      // 玩家进入场景后的角色数据
	LOGIC_TO_CLIENT_CREATURE_DESTORY		= 102;      // 生物销毁
	LOGIC_TO_CLIENT_CREATURE_CREATE			= 103;      // 生物创建
	LOGIC_TO_CLIENT_CREATURE_SYN_ATTR_ALL	= 104;		// 生物属性广播同步
	LOGIC_TO_CLIENT_CREATURE_SYN_ATTR		= 105;      // 生物属性同步单个客户端
	LOGIC_TO_CLIENT_CREATURE_LEVUP          = 106;		// 生物升级
	SERVER_TO_CLIENT_PLAYER_END				= 110;

	//场景
	SERVER_TO_CLIENT_SCENE_BEGIN			= 111;
	LOGIC_TO_CLIENT_ACTOR_MOVE			= 112;      // 生物移动同步
	LOGIC_TO_CLIENT_ACTOR_SKILLMOVE			= 113;      // 技能移动同步
	LOGIC_TO_CLIENT_NOTIFY_LOAD_MAP			= 114;      // 通知客户端加载地图
	LOGIC_TO_CLIENT_SCENE_TRANS_RESULT		= 115;      // 返回场景传送结果
	LOGIC_TO_CLIENT_NOTIFY_TRANS_LOGICNODE		= 116;      // 通知客户端准备换逻辑节点
	LOGIC_TO_CLIENT_ACTOR_BEAT_BACK			= 117;	    //击退
	LOGIC_TO_CLIENT_ACTOR_TELEPORTING		= 118;	    //瞬间移动
	LOGIC_TO_CLIENT_ACTOR_ROCKER_MOVE		= 119;	    //玩家摇杠移动
	LOGIC_TO_CLIENT_ACTOR_POS_UPDATE		= 120;	    //玩家位置更新
	SERVER_TO_CLIENT_SCENE_END				= 129;

	//技能
	SERVER_TO_CLIENT_SKILL_BEGIN			= 130;
    LOGIC_TO_CLIENT_SKILL_RESULT_BROADCAST	= 131;      // 使用技能结果广播
    LOGIC_TO_CLIENT_SKILL_USE_BROADCAST     = 132;      // 生物使用技能广播
	LOGIC_TO_CLIENT_BUFF_RESULT_BROADCAST   = 133;		//生物使用buff结果广播
	LOGIC_TO_CLIENT_SKILL_LEVEL_UP_RSP      = 134;		// 学习技能回复
	LOGIC_TO_CLIENT_SKILL_STOP_USE		= 135;		//技能停止
	SERVER_TO_CLIENT_SKILL_END				= 140;


	//背包
	SERVER_TO_CLIENT_PACKAGE_BEGIN			= 150;
	LOGIC_TO_CLIENT_PACKAGE_LABEL_INFO		= 151;		//背包标签页信息
	LOGIC_TO_CLIENT_PACKAGE_INFO			= 152;		//背包物品信息
	LOGIC_TO_CLIENT_PACKAGE_SORT			= 153;		//整理
	LOGIC_TO_CLIENT_PACKAGE_SELL			= 154;		//出售
	LOGIC_TO_CLIENT_PACKAGE_USE				= 155;		//使用物品
	LOGIC_TO_CLIENT_PACKAGE_UPDATE			= 156;		//背包更新
	LOGIC_TO_CLIENT_PACKAGE_UPDATE_EXPAND	= 157;		//背包更新扩展
	LOGIC_TO_CLIENT_PLAYER_EQUIP_INFO			= 158;		//装备信息
	LOGIC_TO_CLIENT_PLAYER_EQUIP_DRESS			= 159;		//穿戴装备
	LOGIC_TO_CLIENT_PLAYER_EQUIP_UNDRESS		= 160;		//卸下装备
	LOGIC_TO_CLIENT_PLAYER_EQUIP_UPDATE			= 161;		//装备更新
	LOGIC_TO_CLIENT_PLAYER_EQUIP_ONEKEY_DRESS	= 162;		//一键穿戴
	LOGIC_TO_CLIENT_EQUIP_STREN					= 163;		//强化
	LOGIC_TO_CLIENT_EQUIP_ONEKEY_STREN			= 164;		//一键强化
	LOGIC_TO_CLIENT_EQUIP_REFINE				= 165;		//精炼
	LOGIC_TO_CLIENT_EQUIP_DIGHOLE				= 166;		//开孔
	LOGIC_TO_CLIENT_EQUIP_INLAY					= 167;		//镶嵌宝石
	LOGIC_TO_CLIENT_EQUIP_TAKEOFF				= 168;		//取下宝石
	LOGIC_TO_CLIENT_EQUIP_COMPOSE				= 169;		//宝石合成
	LOGIC_TO_CLIENT_EQUIP_RECAST				= 170;		//重铸
	LOGIC_TO_CLIENT_EQUIP_INLAY_HOLY			= 171;		//镶嵌神印石
	LOGIC_TO_CLIENT_EQUIP_TAKEOFF_HOLY			= 172;		//取下神印石
	LOGIC_TO_CLIENT_EQUIP_COMPOSE_MATERIAL		= 173;		//材料合成
	LOGIC_TO_CLIENT_PLAYER_PICKUP_DROP			= 174;		//拾取掉落返回
	LOGIC_TO_CLIENT_PLYAER_EQUIP_DEL			= 175;		//删除角色装备
	SERVER_TO_CLIENT_PACKAGE_END				= 190;
	
	//副本
	SERVER_TO_CLIENT_DUPLICATE_BEGIN			= 200;
	LOGIC_TO_CLIENT_DUPLICATE_INFO				= 201;		//副本信息
	LOGIC_TO_CLIENT_DUPLICATE_ENTER				= 202;		//进入副本
	LOGIC_TO_CLIENT_DUPLICATE_EXIT				= 203;		//退出副本
	LOGIC_TO_CLIENT_DUPLICATE_RESULT			= 204;		//副本结果
	LOGIC_TO_CLIENT_DUPLICATE_AUTO_FIGHT		= 205;		//副本自动战斗
	LOGIC_TO_CLIENT_DUPLICATE_LOTTERY			= 206;		//翻牌
	LOGIC_TO_CLIENT_DUPLICATE_BUY_NUM			= 207;		//购买次数
	LOGIC_TO_CLIENT_DUPLICATE_SWEEP				= 208;		//扫荡
	LOGIC_TO_CLIENT_DUPLICATE_UPDATE			= 209;		//副本更新
	LOGIC_TO_CLIENT_DUPLICATE_RELIVE			= 210;		//复活
	LOGIC_TO_CLIENT_DUPLICATE_LEAVE				= 211;		//离开副本
	LOGIC_TO_CLIENT_DUPLICATE_ENTER_AREA		= 212;		//进入副本区域
	SERVER_TO_CLIENT_DUPLICATE_END				= 220;

	//BUFF
	SERVER_TO_CLIENT_BUFF_BEGIN					= 230;
	LOGIC_TO_CLIENT_START_BUFF_BAG				= 231; //开始状态包
	LOGIC_TO_CLIENT_END_BUFF_BAG				= 232; //结束状态包
	LOGIC_TO_CLIENT_MISS_BUFF_BAG				= 233; //状态包丢失,可能是被免疫了，或叠加不上
	LOGIC_TO_CLIENT_SYN_BUFF_STATE				= 234; //同步特殊BUFF状态
	LOGIC_TO_CLIENT_PLAY_IDLE					= 235; //播放休闲动作
	LOGIC_TO_CLIENT_REPLACE_NORMAL_SKILL		= 236; //替换普工技能
	SERVER_TO_CLIENT_BUFF_END					= 250;
    
	//任务
	SERVER_TO_CLIENT_MISSION_BEGIN				= 270;
	LOGIC_TO_CLIENT_QUERY_MiSSIONLIST			= 271;      // 请求任务列表返回
	LOGIC_TO_CLIENT_ACCEPTMISSION				= 272;		// 接收任务返回
	LOGIC_TO_CLIENT_ABANDONMISSION				= 273;		// 放弃任务返回
	LOGIC_TO_CLIENT_SUBMITMISSION				= 274;		// 提交任务返回
	LOGIC_TO_CLIENT_UPDATEMISSIONSTATUSNOTIFY	= 275;		// 任务状态更新
	LOGIC_TO_CLIENT_UPDATEACCEPTABLELISTNOTIFY	= 276;		// 更新可接任务列表通知
	LOGIC_TO_CLIENT_RECEIVEMISSIONNOTIFY		= 277;      // 接取任务通知
	LOGIC_TO_CLIENT_GCDELETEMISSIONNOTIFY		= 278;      // 删除已接任务通知
	LOGIC_TO_CLIENT_GETCAMPMISSIONINFO			= 279;		// 阵营任务返回
	LOGIC_TO_CLIENT_DICECAMPMISSION				= 280;		// 摇骰子返回
	LOGIC_TO_CLIENT_GETCAMPBOXREWARD			= 281;		// 领取阵营宝箱奖励返回
	SERVER_TO_CLIENT_MISSION_END				= 300;


	//聊天
	SERVER_TO_CLIENT_CHAT_BEGIN				= 310;
	LOGIC_TO_CLIENT_CHATMSGREQ				= 311;		// 聊天返回
	LOGIC_TO_CLIENT_SYSTEMCHATMSGNOTIFY		= 312;		// 系统消息返回
	CENTER_TO_CLIENT_ASKOFFLINEMSG			= 313;		//获取好友离线消息
	CENTER_TO_CLIENT_INTRVALASKOFFLINEMSG	= 314;		//分段获取好友离线消息
	CENTER_TO_CLIENT_CHARACTER_DETAIL_DATA	= 315;		//查看玩家详细数据返回
	SERVER_TO_CLIENT_CHAT_END				= 330;



	//好友模块
	SERVER_TO_CLIENT_FRIEND_BEGIN		 = 350;
	CENTER_TO_CLIENT_APPLY_RELATION		 = 351;  //申请添加好友
	CENTER_TO_CLIENT_AGREE_RELATION		 = 352;  //同意添加好友
	CENTER_TO_CLIENT_DELETE_RELATION	 = 353;  //删除好友
	CENTER_TO_CLIENT_RECENT_CHAT_LIST    = 354;  //请求最近联系人
	CENTER_TO_CLIENT_ONEKEY_FRIEND		 = 355;  //好友推荐
	CENTER_TO_CLIENT_RELATION_INFO		 = 356;  //请求好友信息
	CENTER_TO_CLIENT_SEARCH_USER		 = 357;  //搜索好友
	CENTER_TO_CLIENT_NOTIFY_RELATION     = 358;  //更新好友
	CENTER_TO_CLIENT_GIVE_PHYSICAL		 = 359;  //赠送体力
	CENTER_TO_CLIENT_AllGIVE_PHYSICAL	 = 360;  //一键赠送体力
	CENTER_TO_CLIENT_GET_PHYSICAL		 = 361;  //领取体力
	CENTER_TO_CLIENT_GET_PHYSICAL_GIFT   = 362;  //领取体力礼包
	CENTER_TO_CLIENT_APPLYLIST			 = 363;  //获取好友申请列表
	CENTER_TO_CLIENT_CLEARAPPLYLIST		 = 364;  //清除好友申请列表
	CENTER_TO_CLIENT_PHYSICALNOTIFY		 = 365;  //体力通知
	SERVER_TO_CLIENT_FRIEND_END			 = 380;

	//邮件模块
	SERVER_TO_CLIENT_MAIL_BEGIN				= 400;
	SERVER_TO_CLIENT_SENDMAIL				= 401;
	SERVER_TO_CLIENT_DELETEMAIL				= 402;
	SERVER_TO_CLIENT_NEWMAILNOTIFY			= 403;
	SERVER_TO_CLIENT_GETMAILITEMLIST		= 404;
	SERVER_TO_CLIENT_ASKMAIL				= 405;
	SERVER_TO_CLIENT_UNREADMAILNOTIFY		= 406;
	SERVER_TO_CLIENT_GETMAILATTACHMENT		= 407;
	SERVER_TO_CLIENT_GETALLMAILATTACHMENT	= 408; //一键领取所有邮件附件

	SERVER_TO_CLIENT_MAIL_END			 = 450;

	//阵营
	SERVER_TO_CLIENT_CAMP_BEGIN			 = 460;
	CENTER_TO_CLIENT_CAMP_JOIN			 = 461;		//加入阵营
	CENTER_TO_CLIENT_CAMP_CHANGE		 = 462;		//更换阵营
	CENTER_TO_CLIENT_CAMP_INFO			 = 463;		//阵营信息
	CENTER_TO_CLIENT_CAMP_RANK_INFO		 = 464;		//阵营排名信息
	CENTER_TO_CLIENT_CAMP_HONOR_INFO	 = 465;		//阵营荣誉信息
	CENTER_TO_CLIENT_CAMP_PUBLISH_NOTICE = 466;		//发布阵营公告
	CENTER_TO_CLIENT_CAMP_BROAD_NOTICE	 = 467;		//广播阵营公告
	SERVER_TO_CLIENT_CAMP_END			 = 520;

	//成就
	SERVER_TO_CLIENT_ACHIEVENTMENT_BEGIN		= 530;		
	SERVER_TO_CLIENT_ACHIEVENTMENTNOTIFY		= 531;		//成就改变通知
	SERVER_TO_CLIENT_ACHIEVEMENTINFO			= 532;		//请求成就列表信息返回
	SERVER_TO_CLIENT_GETACHIEVEMENTREWARD		= 534;		//领取单个成就奖励返回
	SERVER_TO_CLIENT_GETACHIEVEMENTBOXREWARD	= 535;		//领取成就宝箱
	SERVER_TO_CLIENT_ACHIEVENTMENT_END			= 540;

	//副业
	SERVER_TO_CLIENT_DEPUTY_BEGIN					= 550;
	SERVER_TO_CLIENT_DEPUTY_COLLECT_SKILL_INFO		= 551;	//请求收集副业技能信息
	SERVER_TO_CLIENT_DEPUTY_MAKE_SKILL_INFO			= 552;	//请求制造副业技能信息
	SERVER_TO_CLIENT_DEPUTY_UPGRADE_COLLECT_SKILL	= 553;	//请求升级收集副业技能
	SERVER_TO_CLIENT_DEPUTY_REFINE_MAKE_SKILL		= 554;	//请求精修制作副业技能
	SERVER_TO_CLIENT_DEPUTY_COLLECT_OPERATE			= 555;	//请求采集副业操作
	SERVER_TO_CLIENT_DEPUTY_MAKE_OPERATE			= 556;	//请求制作副业操作
	SERVER_TO_CLIENT_DEPUTY_NOTIFY_COLLECT			= 557;	//通知采集副业状态
	SERVER_TO_CLIENT_DEPUTY_UPDATE_COLLECT_ENERGY   = 558;	//更新采集副业熟练度



	//PVP
	SERVER_TO_CLIENT_PVP_BEGIN				= 630;
	SERVER_TO_CLIENT_RET_SET_MODE			= 631; //设置PK模式返回
	SERVER_TO_CLIENT_PVP_KILLING_VALUE				= 632; //PVP killing value 广播
	SERVER_TO_CLIENT_UPDATE_RED_PLAYER		= 633; //更新红名列表
	SERVER_TO_CLIENT_PVP_END				= 639;

	//商城
	SERVER_TO_CLIENT_MALL_BEGIN				= 650;
	CENTER_TO_CLIENT_ASK_DATA_RESULT		= 651;	//请求商城数据返回
	LOGIC_TO_CLIENT_BUY_RESULT				= 652;	//购买返回
	CENTER_TO_CLIENT_CHANGE_NOTIFY			= 653;	//通知客户端商城数据变化
	CENTER_TO_CLIENT_GET_TIME_RESULT		= 654;	//返回商品剩余刷新时间
	SERVER_TO_CLIENT_MALL_END				= 660;

	LOGIC_TO_CLIENT_PROTOCOL_COUNT = 1024;

};




enum GAME_STATE 
{
    GAME_STATE_NULL              			= 0;		// 初始状态

    GAME_STATE_LOGIN_ONLINE_BEGIN   		= 1;		// 登陆online server
    GAME_STATE_LOGIN_ONLINE_REPLACE 		= 2;   		// 已在线，等待踢对方下线
    GAME_STATE_LOGIN_ONLINE_END     		= 3;   		// 登陆online server结束

    GAME_STATE_LOAD_CHARLIST_BEGIN   		= 4;		// 开始加载角色列表
    GAME_STATE_LOAD_CHARLIST_END     		= 5;		// 加载角色列表结束

    GAME_STATE_SELECT_ROLE			        = 6;		// 选择角色中
    GAME_STATE_CREATE_ROLE_BEGIN 			= 7;		// 开始创建角色
    GAME_STATE_CREATE_ROLE_END   			= 8;		// 创建角色结束

    GAME_STATE_ENTER_GAME        			= 9;		// 进入游戏中
    GAME_STATE_SYNC_BEGIN        			= 10;		// 同步数据开始
    GAME_STATE_SYNC_END          			= 11;		// 同步数据结束
    GAME_STATE_ENTER_SCENCE      			= 12;		// 进入场景中
    GAME_STATE_PLAYERING         			= 13;		// 正常游戏中
};

enum LOGIN_FLAG 
{
    LOGIN_FLAG_NULL             = 0;					// NULL
    LOGIN_FLAG_NORMAL           = 1;					// 正常登陆
    LOGIN_FLAG_REPLACE          = 2;					// 重复登陆
    LOGIN_FLAG_SWITCH_SERVER    = 3;					// 切换服务器
};

enum LOGOUT_FLAG 
{
    LOGOUT_FLAG_NULL           	= 0;					// NULL
    LOGOUT_FLAG_NORMAL         	= 1;					// 正常下线	
    LOGOUT_FLAG_REPLACE        	= 2;					// 别处登陆，被踢下线	
    LOGOUT_FLAG_SWITCH_SERVER  	= 3;					// 切换服务器下线
	LOGOUT_FLAG_BREAK_TIMEOUT  	= 4;					// 断线超时
};
